app/src/server/tasks/fineTuning/checkFineTuneStatus.task.ts

```diff
@@ -22,7 +27,40 @@ const runOnce = async () => {
       try {
         const resp = await trainingStatus({ callId: job.modalTrainingJobId });
         if (resp.status === "done") {
-          // TODO: kick off job to run test set evals here
+          const fineTune = await prisma.fineTune.findUnique({
+            where: { id: job.id },
+          });
+          if (!fineTune) return;
+          const pruningRules = await prisma.pruningRule.findMany({
```

Comment:
This needs to be extracted and shared between `getCompletion2` and here at least. Also probably with the script that uploads training data. It's way too easy to update the way pruning rules are applied in one place and not another and break things.

Separately, I think there's a potential bug here? Don't pruning rules need to be applied in the same order they're defined or else the results won't necessarily be stable? It doesn't look like this code sorts them before applying them. Haven't checked if we do so in other places but if not I think we need to.

Reply:
> This needs to be extracted and shared between getCompletion2 and here at least. Also probably with the script that uploads training data. It's way too easy to update the way pruning rules are applied in one place and not another and break things.

I'm fine with making the change to pull the strings to prune in a standardized function, but that does mean an extra database call in some contexts, for example `getTestResult.task`, since I've been tacking on the pruning rules to existing queries, which is why it isn't currently standardized.

> Separately, I think there's a potential bug here?

You're right, we weren't ensuring correct order here. I'll make that change.

Reply:
If we ensure the `createdAt` and `id` fields are always queried on the pruning rules we could do the sorting within the `pruneInputMessages` function itself. But the way you have it now is fine.